apps/shipments/views.py

```diff
@@ -77,16 +78,33 @@ def get_tracking_data(self, request, version, pk):
 
         tracking_data = TrackingData.objects.filter(shipment__id=shipment.id)
 
-        if 'as_line' in request.query_params:
-            all_features = build_line_string_feature(shipment, tracking_data)
-        elif 'as_point' in request.query_params:
-            all_features = build_point_features(shipment, tracking_data)
+        if tracking_data.exists() and (not shipment.route_as_point or not shipment.route_as_line):
+            # Update shipment route
+            shipment_route_update.delay(shipment.id)
+
+            if 'as_line' in request.query_params:
```

Comment:
Let's actually remove all of the `as_line` code for now. It is not used by the frontend and it's going to be easier to focus on optimizing `as_point` without `as_line` getting in the way.